pkg/cleaner/aws/aws.go

```diff
@@ -70,22 +78,25 @@ func (a *Cleaner) cleanStacks() error {
 
 	var lastError error
 	for _, stack := range output.Stacks {
-		a.logger.Log("level", "debug", "message", fmt.Sprintf("checking stack %#q", *stack.StackName))
 		if !stackShouldBeDeleted(stack) {
+			a.logger.Log("level", "debug", "message", fmt.Sprintf("leaving stack %#q untouched: %#v", *stack.StackName, *stack))
 			continue
 		}
 		a.logger.Log("level", "debug", "message", fmt.Sprintf("found that stack %#q should be deleted", *stack.StackName))
 
-		enableTerminationProtection := false
-		updateTerminationProtection := &cloudformation.UpdateTerminationProtectionInput{
-			EnableTerminationProtection: &enableTerminationProtection,
-			StackName:                   stack.StackName,
-		}
-		_, err = a.cfClient.UpdateTerminationProtection(updateTerminationProtection)
-		if err != nil {
-			lastError = err
-			// do not return on error, try to continue deleting.
-			a.logger.Log("level", "debug", "message", fmt.Sprintf("failed disabling stack protection %#q: %#v", *stack.StackName, err))
+		if *stack.EnableTerminationProtection {
+			a.logger.Log("level", "debug", "message", fmt.Sprintf("disabling termination protection for stack %#q", *stack.StackName))
+			enableTerminationProtection := false
+			updateTerminationProtection := &cloudformation.UpdateTerminationProtectionInput{
+				EnableTerminationProtection: &enableTerminationProtection,
+				StackName:                   stack.StackName,
+			}
+			_, err = a.cfClient.UpdateTerminationProtection(updateTerminationProtection)
+			if err != nil {
+				lastError = err
+				// do not return on error, try to continue deleting.
+				a.logger.Log("level", "error", "message", fmt.Sprintf("failed disabling stack protection %#q: %#v", *stack.StackName, err))
```

Comment:
from this point we know that an attempt to delete stack will fail because termination protection is still enabled.
How about adding a `continue` here ?

Reply:
yeah!